frame/staking/src/lib.rs

```diff
@@ -1830,7 +1832,7 @@ decl_module! {
 		///     - Write: Payee
 		/// # </weight>
 		#[weight = 11 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(1, 1)]
-		fn set_payee(origin, payee: RewardDestination) {
+		fn set_payee(origin, payee: RewardDestination<T::AccountId>) {
```

Comment:
Does the existing weight take into account that the account may be being created in here?

Reply:
Yes, worst case scenario here is the same as worst case scenario when paying out to controller:

```
		RewardDestination::Controller => Self::bonded(stash)
				.and_then(|controller|
					Some(T::Currency::deposit_creating(&controller, amount))
				),
```

Reply:
And if I remember correctly, bonding the reward is more costly than creating a new account.